src/dmd/dinterpret.d

```diff
@@ -2225,6 +2225,11 @@ public:
             printf("%s DeclarationExp::interpret() %s\n", e.loc.toChars(), e.toChars());
         }
         Dsymbol s = e.declaration;
+        if (auto uda = s.isUserAttributeDeclaration())
+        {
+            assert(uda.decl && uda.decl.dim == 1);
+            s = (*uda.decl)[0];
```

Comment:
Why only one declaration?

Reply:
Currently, at function scope the parser only allows one declaration after UDAs.

Reply:
Fair enough. But the issue should rather be fixed by changing the `AttribDeclaration` branch below (which rejects `enum`s ATM and hence causes the bug)

Reply:
Possible solution (unless im missing something):

```patch
diff --git a/src/dmd/dinterpret.d b/src/dmd/dinterpret.d
index 8676f3b59..3d383b33d 100644
--- a/src/dmd/dinterpret.d
+++ b/src/dmd/dinterpret.d
@@ -2313,12 +2313,8 @@ public:
             AttribDeclaration ad = e.declaration.isAttribDeclaration();
             if (ad && ad.decl && ad.decl.dim == 1)
             {
-                Dsymbol sparent = (*ad.decl)[0];
-                if (sparent.isAggregateDeclaration() || sparent.isTemplateDeclaration() || sparent.isAliasDeclaration())
-                {
-                    result = null;
-                    return; // static (template) struct declaration. Nothing to do.
-                }
+                result = null;
+                return; // static (template) struct declaration. Nothing to do.
             }

             // These can be made to work, too lazy now
```

Reply:
That branch just returns `null`, but if there is a variable declaration inside the UDA it needs to be evaluated.
I kept that code untouched, although I think it should be revisited some day.

Reply:
Ahh, good catch. Probably still better to simply check for any `AttribDeclaration` instead of UDAs:

```patch
diff --git a/src/dmd/dinterpret.d b/src/dmd/dinterpret.d
index 8676f3b59..85c1858a2 100644
--- a/src/dmd/dinterpret.d
+++ b/src/dmd/dinterpret.d
@@ -2225,6 +2225,11 @@ public:
             printf("%s DeclarationExp::interpret() %s\n", e.loc.toChars(), e.toChars());
         }
         Dsymbol s = e.declaration;
+        if (auto ad = s.isAttribDeclaration())
+        {
+            assert(ad.decl && ad.decl.dim == 1);
+            s = (*ad.decl)[0];
+        }
         if (VarDeclaration v = s.isVarDeclaration())
         {
             if (TupleDeclaration td = v.toAlias().isTupleDeclaration())
```

Unless we know for sure that other AD's can never reach this code.